src/netex-convertor/handler.ts

```diff
@@ -61,68 +59,80 @@ const uploadToS3 = async (netex: string, fileName: string): Promise<void> => {
-            const generatedNetex = await netexGen.generate();
-
-            const fileName = generateFileName(s3FileName);
+        const nocs: string[] = buildNocList(ticket);
+        const operatorData = await db.getOperatorDataByNocCode(nocs);
+        const netexGen = netexGenerator(ticket, operatorData);
+        const generatedNetex = await netexGen.generate();
+        const fileName = generateFileName(s3FileName);
 
-            await uploadToS3(generatedNetex, fileName);
+        await uploadToS3(generatedNetex, fileName);
 
-            if (matchingData.nocCode !== 'IWBusCo') {
-                console.info(`NeTEx generation complete for type ${type}`);
-            }
-        } else if (isPeriodTicket(s3Data)) {
-            const userPeriodTicket: PeriodTicket = s3Data;
-            let operatorData: Operator[] = [];
-            if (type === 'multiOperator') {
-                if (isMultiOperatorGeoZoneTicket(userPeriodTicket) || isSchemeOperatorTicket(userPeriodTicket)) {
-                    const nocs: string[] = [...userPeriodTicket.additionalNocs];
-                    if (isMultiOperatorGeoZoneTicket(userPeriodTicket)) {
-                        nocs.push(userPeriodTicket.nocCode);
-                    }
-                    operatorData = await db.getOperatorDataByNocCode(nocs);
-                } else if (isMultiOperatorMultipleServicesTicket(userPeriodTicket)) {
-                    const nocs: string[] = userPeriodTicket.additionalOperators.map(
-                        additionalOperator => additionalOperator.nocCode,
-                    );
-                    nocs.push(userPeriodTicket.nocCode);
-                    operatorData = await db.getOperatorDataByNocCode(nocs);
-                }
-            } else if (
-                !isMultiOperatorGeoZoneTicket(userPeriodTicket) &&
-                !isMultiOperatorMultipleServicesTicket(userPeriodTicket) &&
-                !isSchemeOperatorTicket(userPeriodTicket)
-            ) {
-                operatorData = await db.getOperatorDataByNocCode([userPeriodTicket.nocCode]);
-            }
-            const netexGen = periodTicketNetexGenerator(userPeriodTicket, operatorData);
-            const generatedNetex = await netexGen.generate();
-
-            const fileName = generateFileName(s3FileName);
-
-            await uploadToS3(generatedNetex, fileName);
-
-            if (!isSchemeOperatorTicket(userPeriodTicket) && userPeriodTicket.nocCode !== 'IWBusCo') {
-                console.info(`NeTEx generation complete for type ${type}`);
-            }
-        } else {
-            throw new Error(
-                `The JSON object '${decodeURIComponent(event.Records[0].s3.object.key.replace(/\+/g, ' '))}' in the '${
-                    event.Records[0].s3.bucket.name
-                }' bucket does not contain a 'type' attribute to distinguish product type.`,
-            );
+        if (!isSchemeOperatorTicket(ticket) && ticket.nocCode !== 'IWBusCo') {
```

Comment:
Unsure honestly! It was in the old handler logic. @ljones92 do you know?

Reply:
I think it was probably because we have no way of knowing if a scheme user is an internal user or not like we do with NOC codes. Since we may be adding TFL as a scheme user at some point it would probably be worth removing this check.